File: docs-site/webjs/sdk_onboarding_pub.js
import { CONNECTED, Realtime } from "relayx-webjs"

const client = new Realtime({
  api_key: '$api_key',
  secret: '$secret'
})

client.init()

if(process.argv.length < 4){
    console.log("Usage: node <file>.js <topic> <message>");
    console.log("type = pub or sub")
    process.exit();
}

var topic = process.argv[2]
var message = process.argv[3]

client.connect()

await client.on(CONNECTED, async (status) => {
    console.log("Connected to relayX!")

    console.log(`Sending message to topic => ${topic}`)
    console.log()

    var sent = await client.publish(topic, message);

    console.log(`Message sent => ${sent}`)
    console.log("Exiting script...")

    process.exit(0)
})

 setTimeout(async () => {
    await client.close()
    process.exit(0)
}, 15000)

File: docs-site/python/sdk_onboarding_pub.py
import asyncio
import sys
from relayx_py import Realtime

client = Realtime({
  "api_key": '$api_key',
  "secret": '$secret'
})

client.init()

async def on_connected():
    topic = sys.argv[1]
    message = sys.argv[2]

    print("Connected to relayX!")

    print("Sending message to topic => {}".format(topic))
    print()

    sent = await client.publish(topic, message)

    print(f"Message sent => {sent}")

    print()
    print("Ctrl + C to quit")

async def main():
    if len(sys.argv) < 3:
        print("Usage: python <file name>.py <topic> <message>")

        print()
        print("Ctrl + C to quit")

    topic = sys.argv[1]
    message = sys.argv[2]

    print("Topic => {}".format(topic))
    print("Message => {}".format(message))

    await client.on(Realtime.CONNECTED, on_connected)

    await client.connect() 

# Run the main function
if __name__ == "__main__":
    asyncio.run(main())